### PR TITLE
Add autorefresh in the index page for fullscreen mode feature

### PR DESCRIPTION
**Pull request by ilausuch**
https://progress.opensuse.org/issues/17886

This new feature allows to autorefresh the index page with 30, 60,
120 and 180 seconds. The user controls the interval with the
filter control panel.

### CODE REVIEW

File: templates/webapi/main/index.html.ep [hunk 1]
```diff
@@ -81,7 +81,7 @@
                 %>
             </div>
 
-            <div class="form-group">
+            <div class="form-group" id="filter-interval-content">
```
Comment (Martchus) [inline templates/webapi/main/index.html.ep:84]: When fixing things here anyways, could you also take care that the `input` elements contained by this `div` have unique IDs?

**Comment by Martchus** [comment]
Could you start the PR title and commit messages with a verb in imperative so it is clear whether this is a fix/extension/improvement of an existing feature or a new feature.

File: templates/webapi/main/index.html.ep (hunk 1 above)
Comment (ilausuch) [reply templates/webapi/main/index.html.ep:84]: Yes, you're right. Fixed

**Comment by ilausuch** [comment]
Also is fixed a bug in the code that didn't show the interval selector when refresh the page and the full screen mode was on

File: templates/webapi/main/index.html.ep [hunk 2]
```diff
@@ -80,6 +80,25 @@
...
             </div>
+
+            <div class="form-group" id="filter-interval-content">
+                <strong>
+                    Auto refresh
+                    <%= help_popover('Help for <i>Auto refresh</i>' => '<p>Allows to update the data of this page automatically</p>') %>
+                </strong>
+
+                <input value="0" name="interval" type="radio">
+                <label for="filter-interval">no refresh</label>
```
Comment (Martchus) [inline templates/webapi/main/index.html.ep:91]: But now the `for` attributes don't refer to a valid ID anymore.

Comment (ilausuch) [reply templates/webapi/main/index.html.ep]: yes, new commit incomming

File: templates/webapi/main/index.html.ep [hunk 3]
```diff
@@ -80,6 +80,25 @@
...
+                <input value="0" name="interval" type="radio" id="filter-interval_0">
+                <label for="filter-interval_0">no refresh</label>
+                <input value="30" name="interval" type="radio" id="filter-interval_30">
+                <label for="filter-interval_30">30 seconds</label>
+                <input value="60" name="interval" type="radio" id="filter-interval_60">
+                <label for="filter-interval_60">one minute</label>
+                <input value="120" name="interval" type="radio" id="filter-interval_120">
+                <label for="filter-interval_120">two minutes</label>
+                <input value="180" name="interval" type="radio" id="filter-interval_180">
+                <label for="filter-interval_180">tree minutes</label>
```
Comment (Martchus) [inline templates/webapi/main/index.html.ep:99]: s/tree/three/

File: assets/javascripts/index.js [hunk 4]
```diff
@@ -18,6 +18,9 @@ function setupIndexPage() {
         showTagsCheckBox.prop('disabled', checked);
     });
     defaultExpanedCheckBox.prop('checked', false);
+    filterFullScreenCheckBox.on('change', function(){
```
Comment (Martchus) [inline assets/javascripts/index.js:21]: I'd still prefer `() {` over `(){`.

File: assets/javascripts/index.js [hunk 5]
```diff
@@ -113,3 +118,18 @@ function loadBuildResults(queryParams) {
         },
     });
 }
+
+function autoRefreshRestart(){
```
Comment (Martchus) [inline assets/javascripts/index.js:122]: I'd still prefer `() {` over `(){`.

File: assets/javascripts/index.js [hunk 6]
```diff
@@ -113,3 +118,18 @@ function loadBuildResults(queryParams) {
...
+    if (window.autoreloadIntervalId)
+        clearInterval(window.autoreloadIntervalId);
+
+    if (window.autoreload > 0)
+        window.autoreloadIntervalId = setInterval(loadBuildResults, window.autoreload*1000);
```
Comment (Martchus) [inline assets/javascripts/index.js:127]: We usually surround operators with white spaces, so for consistency…

File: assets/javascripts/index.js [hunk 7]
```diff
@@ -113,3 +118,18 @@ function loadBuildResults(queryParams) {
...
+}
+
+function updateFilterAutoRefreshVisibility(){
```
Comment (Martchus) [inline assets/javascripts/index.js:130]: I'd still prefer `() {` over `(){`.

File: templates/webapi/main/index.html.ep [hunk 8]
```diff
@@ -80,6 +80,25 @@
...
                 %>
             </div>
+
+            <div class="form-group" id="filter-interval-content">
+                <strong>
+                    Auto refresh
+                    <%= help_popover('Help for <i>Auto refresh</i>' => '<p>Allows to update the data of this page automatically</p>') %>
+                </strong>
+
+                <input value="0" name="interval" type="radio" id="filter-interval_0">
```
Comment (kalikiana) [inline templates/webapi/main/index.html.ep:90]: Can we avoid mixing hyphens and underscores? I don't care that much but having both is confusing.

**Comment by ilausuch** [comment]
Ok, I am removing the part that only allows auto-refresh in fullscreen mode

**Comment by okurz** [comment]
your tests fail now in the cache step as the versions of packages can not be found anymore, see https://app.circleci.com/pipelines/github/os-autoinst/openQA/3181/workflows/7eeb8bf3-ee48-433d-a525-943e06cc17a6/jobs/30369/steps

@kalikiana this makes me think that what is tested is actually *not* the to-be-merged state but pre-merge.

@ilausuch please try to fix that by rebasing on a current master with updated dependencies.txt

**Comment by ilausuch** [comment]
Now is not necessary to be in a fullscreen mode to have auto-refresh. Also I rebased.
There are 4 commits, when you approve this, I'll do squash before merge

**Comment by okurz** [comment]
t/ui/16-tests_job_next_previous.t failed. I doubt your changes should influence that though. Please squash the commits and we will see new test results anyway

**Comment by Martchus** [comment]
I've re-triggered only the failing UI tests. That should be faster. There's no need to squash commits locally anymore.

**Pull request merged**